lib/administrate/search.rb

```diff
@@ -6,42 +6,38 @@ class Search
     def initialize(scoped_resource, dashboard_class, term)
       @dashboard_class = dashboard_class
       @scoped_resource = scoped_resource
-      @term = term
+      @term = term || ""
     end
 
     def run
       if @term.blank?
         @scoped_resource.all
       else
-        @scoped_resource.where(query, *search_terms)
+        queries, search_terms = query.transpose
+        @scoped_resource.where(queries.join(" OR "), *search_terms)
       end
     end
 
+    def available?
+      !query.empty?
+    end
+
     private
 
     def query
-      search_attributes.map do |attr|
+      @query ||= attribute_types.keys.map do |attr|
         table_name = ActiveRecord::Base.connection.
           quote_table_name(@scoped_resource.table_name)
         attr_name = ActiveRecord::Base.connection.quote_column_name(attr)
```

Comment:
Do you think so? I take your point about it being less obvious (so perhaps there is another solution) but the advantage of this approach is that the burden of quoting queries doesn't fall on the field author. Since adding fields is one of the easiest way to extend administrate for new users, it feels a little safer to centralise this.

What do you think?

Reply:
Yes, that thought crossed my mind. The solution may be in adding a layer of indirection there, an intermediate method perhaps...? I'm busy today and won't be able to do this just yet, but it's on my radar.